### PR TITLE
Add GenContainer tool script

### PR DESCRIPTION
**Pull request by mauricema**
This patch added GenContainer.py script to reate, extrace, replace,
sign and display a container image. If platform provides
GetContainerList() in BoardConfig.py, the build process will consume
it to create container images.

Signed-off-by: Maurice Ma <maurice.ma@intel.com>

### CODE REVIEW

**Comment by myguitar** [comment]
typo in commit message - to reate, extrace

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 1]
```diff
@@ -0,0 +1,591 @@
+#!/usr/bin/env python
+## @ GenContainer.py
+# Tools to operate on a container image
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: yes, the same tool should be able to do that.

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+	_pack_ = 1
+	_fields_ = [
+		('name',				ARRAY(c_char, 4)),	 # SBL pod entry name
+		('offset',			c_uint32),	 # Component offset in byte from the payload (data)
+		('size',				c_uint32),	 # Region/Component size in byte
+		('reserved',		c_uint8),		 # reserved
+		('alignment',		c_uint8),		 # This image need to be loaded to memory	 in (1 << Alignment) address
+		('auth_type',		c_uint8),		 # Refer AUTH_TYPE_VALUE: 0 - "NONE"; 1- "SHA2_256";	2- "RSA2048SHA256"; 3- "SHA2_384"; 4 - RSA3072SHA384
+		('hash_size',		c_uint8)		 # Hash data size, it could be image hash or public key hash
+		]
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+		('signature',		 ARRAY(c_char, 4)), # Identifies structure
+		('version',			 c_uint16),					# Header version
+		('data_offset',	 c_uint16),					# Offset of payload (data) from header in byte
+		('data_size',		 c_uint32),					# Size of payload (data) in byte
+		('auth_type',		 c_uint8),					# Refer AUTH_TYPE_VALUE: 0 - "NONE"; 2- "RSA2048SHA256"; 4 - RSA3072SHA384
+		('image_type',	 c_uint8),					#	 03 - POD_TYPE_CLASSIC //Files: cmdline, bzImage, initrd, acpi, firmware1, firmware2, ...
+																				#	 04 - POD_TYPE_MULTIBOOT // cmdline1, elf1, cmdline2, elf2, ...
+		('flags',				 c_uint8),					# #define SBL_POD_FLAG_IMAGE BIT0	 #define SBL_POD_FLAG_COMPRESSED	- BIT1
+		('entry_count',	 c_uint8),					# Number of entry in the header
+		]
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+
+	def __new__(cls, buf = None):
+		if buf is None:
+			return Structure.__new__(cls)
+		else:
+			return cls.from_buffer_copy(buf)
+
+	def __init__(self, buf = None):
+		self.priv_key		= ''
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it

File: BootloaderCorePkg/Tools/GenContainer.py [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+
+class CONTAINER ():
+	_struct_display_indent = 18
+	_auth_type_value = {
+			"NONE"				: 0,
+			"SHA2_256"		: 1,
+			"SHA2_384"		: 2,
+			"RSA2048"			: 3,
+			"RSA3072"			: 4,
+		}
```
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it

File: BootloaderCorePkg/Tools/GenContainer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: > typo in commit message - to reate, extrace

Thanks, will fix it.

File: BootloaderCorePkg/Tools/GenContainer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it.

File: BootloaderCorePkg/Tools/GenContainer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it

File: BootloaderCorePkg/Tools/GenContainer.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mauricema) [reply BootloaderCorePkg/Tools/GenContainer.py]: will fix it

**Comment by mauricema** [comment]
> typo in commit message - to reate, extrace

Thanks,  will fix the commit message

**Comment by mauricema** [comment]
Uploaded new PR and addressed all comments.

**Pull request merged**